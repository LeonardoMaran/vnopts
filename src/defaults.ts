import { apiDescriptor } from './descriptors/api';
import { commonDeprecatedHandler } from './handlers/deprecated/common';
import { commonInvalidHandler } from './handlers/invalid';
import { levenUnknownHandler } from './handlers/unknown/leven';
import {
  DeprecatedHandler,
  Descriptor,
  InvalidHandler,
  UnknownHandler,
} from './types';

export const defaultDescriptor: Descriptor = apiDescriptor;
export const defaultUnknownHandler: UnknownHandler = levenUnknownHandler;
export const defaultInvalidHandler: InvalidHandler = commonInvalidHandler;
export const defaultDeprecatedHandler: DeprecatedHandler = commonDeprecatedHandler;
