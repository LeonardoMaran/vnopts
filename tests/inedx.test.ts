import * as vnopts from '../src';
import { createLogger } from './__helpers__/utils';

const logger = createLogger();

beforeEach(() => {
  logger.clearMessages();
});

const schemas = [
  vnopts.createSchema(vnopts.BooleanSchema, {
    name: 'useFlowParser',
    deprecated: true,
    redirect: (value: boolean) =>
      !value ? undefined : { to: { key: 'parser', value: 'flow' } },
  }),
  vnopts.createSchema(vnopts.ChoiceSchema, {
    name: 'parser',
    validate: (value: unknown) =>
      ['function', 'string'].indexOf(typeof value) !== -1,
    choices: [
      'flow',
      'babylon',
      'typescript',
      {
        value: 'postcss',
        deprecated: true,
        redirect: { key: 'parser', value: 'css' },
      },
      'css',
      'less',
      'scss',
    ],
  }),
];

test('deprecated', () => {
  expect(
    vnopts.normalize({ useFlowParser: true }, schemas, { logger }),
  ).toMatchSnapshot();
  expect(logger.getMessages()).toMatchSnapshot();
});

test('redirect', () => {
  expect(
    vnopts.normalize({ parser: 'postcss' }, schemas, { logger }),
  ).toMatchSnapshot();
  expect(logger.getMessages()).toMatchSnapshot();
});
