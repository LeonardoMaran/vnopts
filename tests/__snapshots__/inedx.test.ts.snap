// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deprecated 1`] = `
Object {
  "parser": "flow",
}
`;

exports[`deprecated 2`] = `
Array [
  "useFlowParser is deprecated, now treats it as { parser: \\"flow\\" }.",
]
`;

exports[`redirect 1`] = `
Object {
  "parser": "css",
}
`;

exports[`redirect 2`] = `
Array [
  "{ parser: \\"postcss\\" } is deprecated, now treats it as { parser: \\"css\\" }.",
]
`;
